File: perm_missing_elem/perm_missing_elem.py
# https://app.codility.com/demo/results/trainingXGYGGP-FVX/


def solution(A):
    if (not len(A)):
        return 1

    A.sort()
    result = len(A) + 1

    for i in range(0, len(A)):
        if (i + 1 != A[i]):
            result = i + 1
            break

    return result

File: perm_missing_elem/perm_missing_elem.js
// https://app.codility.com/demo/results/trainingGB8KCJ-3ZC/

function solution(A) {
  if (A.length < 1) {
    return 1;
  }

  const sorted = A.sort((a, b) => a - b);
  let result = A.length + 1;

  for (let i = 1; i <= A.length; i++) {
    if (i !== sorted[i - 1]) {
      result = i;
      break;
    }
  }

  return result;
}
